### ISSUE TITLE
[BUG] onAnimationComplete never fires with children with looping animations

### ISSUE THREAD

**Issue by evenmed** [framer/motion#519]
**Describe the bug**

When a `motion` component has a child which has `yoyo: Infinity`(looping animation), onAnimationComplete never fires.

**IMPORTANT: Provide a CodeSandbox reproduction of the bug**

https://codesandbox.io/s/youthful-tree-sp3nh?fontsize=14&hidenavigation=1&theme=dark

**Steps to reproduce**
1. The CodeSandBox has 2 `motion` components: one has a child with a looping animation and the other doesn't
2. They both have an `onAnimationComplete` prop to trigger an alert
3. Only the component without a child with a looping animation ever fires

**Expected behavior**

I understand that you might want to wait until the amount of repetitions specified in `yoyo` have happened before firing `onAnimationComplete`, but I think when the value is Infinity it should be ignored.

**Comment by mattgperry** [comment]
I disagree that `onAnimationComplete` should ever fire if `repeat: Infinity` is set but I can see the use of an `onAnimationRepeat` prop
